litmus-portal/backend/subscriber/subscriber.go

```diff
@@ -1,117 +1,75 @@
+	newKey string
+)
 
 func init() {
-	if IsValidUrl(GRAPHQL_SERVER_ADDRESS) == false {
-		log.Fatal("GRAPHQL SERVER ADDRESS NOT FOUND OR NOT CORRECT")
+	if home := homeDir(); home != "" {
+		k8s.KubeConfig = flag.String("kubeconfig", filepath.Join(home, ".kube", "config"), "(optional) absolute path to the kubeconfig file")
+	} else {
+		k8s.KubeConfig = flag.String("kubeconfig", "", "absolute path to the kubeconfig file")
 	}
-}
-
-func main() {
+	flag.Parse()
 
-	kubeconfig = flag.String("kubeconfig", "", "absolute path to the kubeconfig file")
-
-	client := &http.Client{}
-	query := `{
-			"query": "subscription { clusterSubscription { id data }  }"
-		}`
-
-	req, err := http.NewRequest("POST", GRAPHQL_SERVER_ADDRESS+"/query", strings.NewReader(query))
+	var isConfirmed bool
+	isConfirmed, newKey, err = cluster.IsClusterConfirmed(clusterData)
 	if err != nil {
 		log.Fatal(err)
 	}
 
-	// Headers for the calling the server endpoint
-	req.Header.Set("Accept-Encoding", "gzip, deflate, br")
-	req.Header.Set("Content-Type", "application/json")
-	req.Header.Set("Accept", "application/json")
-	req.Header.Set("Connection", "keep-alive")
-	req.Header.Set("DNT", "1")
-
-	for {
-		resp, err := client.Do(req)
-		if err != nil || resp.StatusCode != 200 {
-			log.Fatal(err)
-		}
+	if isConfirmed == true {
+		clusterData["KEY"] = newKey
+	} else if isConfirmed == false {
 
-		bodyText, err := ioutil.ReadAll(resp.Body)
+		bodyText, err := gql.ClusterConfirm(clusterData)
 		if err != nil {
 			log.Fatal(err)
 		}
 
-		/*
-			The format of response is data ---map--> clusterSubsription --map--> data(string)
-			- Unmarshal the byte response and store it in a interface
-		*/
 		var responseInterface map[string]map[string]map[string]interface{}
-		err = json.Unmarshal([]byte(bodyText), &responseInterface)
-		if err != nil || len(responseInterface) == 0 {
-			log.Fatal(err)
-		}
-		/*
-			We need to unmarshal it again because in the last map reponseInterface i.e., data is in string type
-		*/
-		var dataInterface map[string]interface{}
-		err = json.Unmarshal([]byte(fmt.Sprint(responseInterface["data"]["clusterSubscription"]["data"])), &dataInterface)
-		if err != nil || len(dataInterface) == 0 {
-			log.Fatal(err)
-		}
-
-		// Calling clusterOperations Function to apply the manifest in the k8s cluster
-		responseFromCluster, err := cluster.ClusterOperations(dataInterface, kubeconfig)
+		err = json.Unmarshal(bodyText, &responseInterface)
 		if err != nil {
 			log.Fatal(err)
 		}
 
-		log.Println(responseFromCluster)
-
-		/*
-			Send response of subscriber to the graphql server as a mutation
-		*/
+		if responseInterface["data"]["clusterConfirm"]["isClusterConfirmed"] == true {
+			log.Println("cluster confirmed")
+			clusterData["KEY"] = strings.TrimSpace(responseInterface["data"]["clusterConfirm"]["newClusterKey"].(string))
+			cluster.ClusterRegister(clusterData)
+		} else {
+			log.Fatal("Cluster not confirmed")
+		}
+	}
+}
 
+func homeDir() string {
```

Comment:
This one is for out-cluster config right? Not very sure if this will work if someone sets the kubeconfig path to a different dir. What I did was just took the whole kubeconfig path as an env variable.

Reply:
Not we don't have to pass kubeconfig,

Reply:
Generally, In outcluster, the path /home/user/.kube/config

Reply:
Outcluster is just for local development

Reply:
Actually, both is present, if you don't pass kubeconfig. It will take it from default config path. else it will take it from the flag